src/api/routes/weather.py

```
@@ -0,0 +1,30 @@
+from fastapi import APIRouter, HTTPException
+from typing import List
+from src.database.models import WeatherData
+from src.database.connection import get_db
+from sqlalchemy.orm import Session
+
+router = APIRouter()
+
+@router.get("/weather", response_model=List[WeatherData])
+def get_weather_data(db: Session = next(get_db())):
```

Ruff (0.12.2)

Do not perform function call `next` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)